### PR TITLE
RDK-35695: Implement group of thunder APIs for CS plugin

### PR DESCRIPTION
**Pull request by praveenrajkg**

### CODE REVIEW

File: AVOutput/AVOutput.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
```
Comment (apatel859) [inline AVOutput/AVOutput.cpp:5]: 2023

File: AVOutput/AVOutput.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    AVOutput::~AVOutput()
+    {
+        LOGINFO();
+    }
+
+    const std::string AVOutput::Initialize(PluginHost::IShell* service)
+    {
+	LOGINFO("Entry\n");
```
Comment (apatel859) [inline AVOutput/AVOutput.cpp:44]: can we move entry exit logs to debug level

File: AVOutput/AVOutput.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
```
Comment (apatel859) [inline AVOutput/AVOutput.h:5]: 2023

File: AVOutput/AVOutputBase.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    private:
+        AVOutputBase(const AVOutputBase&) = delete;
+        AVOutputBase& operator=(const AVOutputBase&) = delete;
+
+   public:
+        AVOutputBase();
+        ~AVOutputBase();
+    public:
+        uint8_t _skipURL;
```
Comment (apatel859) [inline AVOutput/AVOutputBase.h:53]: what is this skipURL for?

File: AVOutput/AVOutputTV.cpp [hunk 5]
```diff
@@ -0,0 +1,7556 @@
...
+	registerMethod("getHLGModeCaps", &AVOutputTV::getHLGModeCaps, this);
+        registerMethod("getSupportedHDR10Modes", &AVOutputTV::getSupportedHDR10Modes, this);
+	registerMethod("getHDR10Mode", &AVOutputTV::getHDR10Mode, this);
+	registerMethod("setHDR10Mode", &AVOutputTV::setHDR10Mode, this);
+	registerMethod("resetHDR10Mode", &AVOutputTV::resetHDR10Mode, this);
+	registerMethod("getHDR10ModeCaps", &AVOutputTV::getHDR10ModeCaps, this);
+	registerMethod("getVideoFormat", &AVOutputTV::getVideoFormat, this);
+	registerMethod("getVideoSource", &AVOutputTV::getVideoSource, this);
+	registerMethod("getVideoFrameRate", &AVOutputTV::getVideoFrameRate, this);
+	registerMethod("getVideoResolution", &AVOutputTV::getVideoResolution, this);
```
Comment (apatel859) [inline AVOutput/AVOutputTV.cpp:483]: Are this method from dispaysetting? if so why only for TV?

File: AVOutput/AVOutputTV.cpp [hunk 6]
```diff
@@ -0,0 +1,7556 @@
...
+                break;
+            }
+        }
+        return index;
+    }
+
+    void AVOutputTV::NotifyVideoFormatChange(tvVideoHDRFormat_t format)
+    {
+        JsonObject response;
+        response["currentVideoFormat"] = getVideoFormatTypeToString(format);
```
Comment (apatel859) [inline AVOutput/AVOutputTV.cpp:318]: is this come from TVsetting ?

File: AVOutput/AVOutputTV.cpp [hunk 7]
```diff
@@ -0,0 +1,7556 @@
...
+            returnResponse(true);
+        }
+    }
+
+    uint32_t AVOutputTV::getZoomMode(const JsonObject& parameters, JsonObject& response)
+    {
+
+        LOGINFO("Entry\n");
+        tvDisplayMode_t mode;
+
```
Comment (apatel859) [inline AVOutput/AVOutputTV.cpp:953]: im not reviwing this implemetation in detials assuming it directly come from TV settings plugins

File: AVOutput/AVOutput.cpp (hunk 1 above)
Comment (praveenrajkg) [reply AVOutput/AVOutput.cpp:5]: ok will change it

File: AVOutput/AVOutput.cpp (hunk 2 above)
Comment (praveenrajkg) [reply AVOutput/AVOutput.cpp:44]: We need that log to capture the API entry and Exit point for debug purpose.

File: AVOutput/AVOutput.h (hunk 3 above)
Comment (praveenrajkg) [reply AVOutput/AVOutput.h:5]: ok will change it

File: AVOutput/AVOutputBase.h (hunk 4 above)
Comment (praveenrajkg) [reply AVOutput/AVOutputBase.h:53]: It Came from tvsettings-plugin.
Its not at all used .We Will remove it.

File: AVOutput/AVOutputTV.cpp (hunk 6 above)
Comment (praveenrajkg) [reply AVOutput/AVOutputTV.cpp:315]: Yes. It is a tvsettings API.
void TVMgr::NotifyVideoFormatChange(tvVideoHDRFormat_t format)

File: AVOutput/AVOutputTV.cpp (hunk 7 above)
Comment (praveenrajkg) [reply AVOutput/AVOutputTV.cpp:950]: API comes from tvsettings-plugin. But entire logic changed as per 
 https://etwiki.sys.comcast.net/display/RDKV/AVOutput+Plugin#AVOutputPlugin

File: AVOutput/AVOutputTV.cpp [hunk 8]
```diff
@@ -0,0 +1,7418 @@
...
+        }
+
+    }
+
+    uint32_t AVOutputTV::resetContrast(const JsonObject& parameters, JsonObject& response)
+    {
+
+        LOGINFO("Entry\n");
+
+        std::string value;
```
[the remark this answers is not in the document]
Comment (praveenrajkg) [reply AVOutput/AVOutputTV.cpp:1780]: std::string by default intilized in it's string class constructor.

File: AVOutput/AVOutputTV.cpp [hunk 9]
```diff
@@ -0,0 +1,7418 @@
...
+        std::string pqmode;
+        std::string source;
+        std::string format;
+        int sourceIndex=0,pqIndex=0,formatIndex=0,brightness=0;
+        int params[3]={0};
+        tvError_t ret = tvERROR_NONE;
+
+        if (parsingSetInputArgument(parameters, "Brightness",source, pqmode, format) != 0)
+        {
+            LOGERR("%s: Failed to parse the input arguments \n", __FUNCTION__);
```
[the remark this answers is not in the document]
Comment (praveenrajkg) [reply AVOutput/AVOutputTV.cpp:1578]: We handled exit scenario only in success case

**Pull request closed without merging**